flexget/plugins/operate/free_space.py

```diff
@@ -1,19 +1,59 @@
 import shutil
-from pathlib import Path
+from enum import Enum
+from pathlib import Path, PurePath
 
+from datamodel_code_generator.util import field_validator, model_validator
 from loguru import logger
+from pydantic import BaseModel, ByteSize, ConfigDict
 
 from flexget import plugin
-from flexget.config_schema import parse_size
 from flexget.event import event
 
 logger = logger.bind(name='free_space')
 
-ABORT_BELOW = 'below'
-ABORT_ABOVE = 'above'
 
-
-def get_free_space(config, task):
+class AbortMode(str, Enum):
+    BELOW = 'below'
+    ABOVE = 'above'
+
+
+class Config(BaseModel):
+    space: ByteSize | float
+    abort_if: AbortMode = AbortMode.BELOW
+    path: PurePath = None
+    port: int = 22
+    host: str = None
+    user: str = None
+    ssh_key_filepath: str = None
+    allotment: ByteSize | float = None
+
+    @field_validator('space', 'allotment', mode='wrap')
+    @classmethod
+    def convert_to_mb(cls, v, handler):
+        # Original config had these fields as MiB, but using ByteSize returns bytes if the field was a str
+        r = handler(v)
+        if not isinstance(v, (int, float)):
+            return r / 1024 / 1024
+        return r
+
+    @model_validator(mode='before')
+    @classmethod
+    def standardize(cls, data):
+        """Converts to the full form of the config from simplified form."""
+        if isinstance(data, (float, int)):
+            data = {'space': data}
+        return data
+
+    model_config = ConfigDict(
+        extra='forbid',
+        json_schema_extra={
+            "type": ["object", "string", "number"],
+            "dependentRequired": {"host": ["user", "ssh_key_filepath", "path"]},
```

Comment:
I don't really love customizing the json schema here, but it is nicer than above, as we are properly describing our schema input differently than our validated output.